cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_brute_force_dpc.cpp

```diff
@@ -0,0 +1,97 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#define DAAL_SYCL_INTERFACE
+#define ONEAPI_DAL_DATA_PARALLEL 1
+
+#include "data_management/data/numeric_table.h"
+#include "oneapi/dal/backend/interop/table_conversion.hpp"
+#include "src/algorithms/k_nearest_neighbors/oneapi/bf_knn_classification_predict_kernel_ucapi.h"
+
+#include "oneapi/dal/algo/knn/backend/gpu/infer_kernel.hpp"
+#include "oneapi/dal/algo/knn/backend/model_interop.hpp"
+#include "oneapi/dal/algo/knn/detail/model_impl.hpp"
+#include "oneapi/dal/backend/interop/common.hpp"
+#include "oneapi/dal/backend/interop/common_dpc.hpp"
+#include "oneapi/dal/backend/interop/error_converter.hpp"
+#include "oneapi/dal/detail/common.hpp"
+
+namespace oneapi::dal::knn::backend {
+
+using dal::backend::context_gpu;
+using namespace daal::data_management;
+
+namespace daal_knn = daal::algorithms::bf_knn_classification;
+namespace interop  = dal::backend::interop;
+
+template <typename Float>
+using daal_knn_brute_force_kernel_t =
+    daal_knn::prediction::internal::KNNClassificationPredictKernelUCAPI<Float>;
+
+template <typename Float>
+static infer_result call_daal_kernel(const context_gpu& ctx,
+                                     const descriptor_base& desc,
+                                     const table& data,
+                                     model m) {
+    auto queue = ctx.get_queue();
+    interop::execution_context_guard guard(queue);
+
+    const std::int64_t row_count    = data.get_row_count();
+    const std::int64_t column_count = data.get_column_count();
+
+    auto arr_data   = row_accessor<const Float>{ data }.pull();
+    auto arr_labels = array<Float>::empty(1 * row_count);
```

Comment:
If you don't specify queue in such operations, computations will be performed on host.
```suggestion
    auto arr_data   = row_accessor<const Float>{ data }.pull(queue);
    auto arr_labels = array<Float>::empty(queue, 1 * row_count);
```